### PR TITLE
LLM Integration

### PR DESCRIPTION
**Pull request by bigabig**
- add ollama to docker-compose
- add ollama client as new dependency for backend / celery worker (we need to rebuild containers)
- adds new redis channel for llm tasks
- adds endpoints for llm tasks
- new feature: llm assistant for doc tagging

### CODE REVIEW

**Comment by bigabig** [comment]
Todos:
- [ ] LLM Annotation in LLM Assistant
- [ ] LLM Metadata Extraction in LLM Assistant
- [ ] LLM Paraphrasing / Summarization as new option when annotating
- [ ] LLM Jobs Manager (Similar to the Pre-pro jobs / Background Jobs UI
- [ ] Delete finished LLM Jobs
- [ ] (maybe?) Introduce device id env variable

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -210,6 +214,26 @@ services:
...
+    volumes:
+      - ./ollama_cache:/root/.ollama
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - driver: nvidia
+              device_ids: ["0"]
+              capabilities: [gpu]
+    networks:
```
Comment (floschne) [inline docker/docker-compose.yml:234]: I would add a docker profile "ollama" similar to the "ray" profile to control whether the container should be started or not

File: backend/src/app/core/data/llm/llm_service.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        if "\n" not in response:
+            return [], "The answer has to contain a line break."
+
+        components = re.split(r"\n+", response)
+
+        # check that the answer contains at least 2 lines
+        if len(components) < 2:
+            return [], "The answer has to contain at least 2 lines."
+
+        # check that the answer starts with "Kategorie:"
```
Comment (floschne) [inline backend/src/app/core/data/llm/llm_service.py:179]: Is German on purpose here?

**Review by floschne — CHANGES_REQUESTED** [review]
Please add `OllamaService` and `LLMService` initializations to the `startup.py::init_services` method :)

**Pull request merged**